### PR TITLE
Add XCBuildConfiguration.append public method

### PR DESCRIPTION
**Pull request by pepibumur**
Resolves https://github.com/tuist/xcodeproj/issues/395

### Short description 📝
Add public method to conveniently append settings as suggested [here](https://github.com/tuist/xcodeproj/issues/395)

### Implementation 👩‍💻👨‍💻
- [x] Implement method.
- [x] Add unit tests.

### CODE REVIEW

**Comment by pepibumur** [comment]
Note that some of the changes of the diff were generated after running the code formatters.

File: Tests/xcodeprojTests/Objects/Configuration/XCBuildConfigurationTests.swift [hunk 1]
```diff
@@ -18,6 +18,47 @@ final class XCBuildConfigurationTests: XCTestCase {
...
+        // Given
+        let subject = XCBuildConfiguration(name: "Debug",
+                                           baseConfiguration: nil,
+                                           buildSettings: ["OTHER_LDFLAGS": "flag1"])
+
+        // When
+        subject.append(setting: "OTHER_LDFLAGS", value: "flag2")
+
+        // Then
+        XCTAssertEqual(subject.buildSettings["OTHER_LDFLAGS"] as? String, "flag1 flag2")
```
Comment (kwridan) [inline Tests/xcodeprojTests/Objects/Configuration/XCBuildConfigurationTests.swift]: I noticed this failed locally for me

```swift
XCTAssertEqual failed: ("Optional("flag2 flag1")") is not equal to ("Optional("flag1 flag2")")
```

It's due to using a Set to reconstruct the setting. I think this test itself is correct as it is, appending settings should append the new setting to the end.

File: Tests/xcodeprojTests/Objects/Configuration/XCBuildConfigurationTests.swift [hunk 2]
```diff
@@ -18,6 +18,47 @@ final class XCBuildConfigurationTests: XCTestCase {
...
+                                               "PRODUCT_NAME": "$(TARGET_NAME:c99extidentifier)",
+                                           ])
+
+        // When
+        subject.append(setting: "PRODUCT_NAME", value: "$(TARGET_NAME:c99extidentifier)")
+
+        // Then
+        XCTAssertEqual(subject.buildSettings["PRODUCT_NAME"] as? String, "$(TARGET_NAME:c99extidentifier)")
+    }
+
```
[the remark this answers is not in the document]
Comment (pepibumur) [reply Tests/xcodeprojTests/Objects/Configuration/XCBuildConfigurationTests.swift]: Handled!

**Comment by pepibumur** [comment]
@kwridan I've added logic to handle the settings that are arrays of strings. I added the test that you proposed to test that it's properly handled.

Regarding booleans, I think it should be fine because they are stored as strings.

Notice in the changes that in the case of arrays, I'm removing the duplicates. That's not straightforward in the case of strings. I was thinking though about removing duplicated `$(inherited)`'s . What do you think?

**Comment by kwridan** [comment]
Thanks @pepibumur 

> Notice in the changes that in the case of arrays, I'm removing the duplicates. That's not straightforward in the case of strings. I was thinking though about removing duplicated $(inherited)'s . What do you think?

That's fair - perhaps we can just remove duplicates as a whole e.g. 

`case let string as String where string != value:`

This will protect against the case where you append the same item twice.

```swift
func test_append_when_duplicateSettingExists() {
        // Given
        let subject = XCBuildConfiguration(name: "Debug",
                                           baseConfiguration: nil,
                                           buildSettings: ["OTHER_LDFLAGS": "flag1"])
        
        // When
        subject.append(setting: "OTHER_LDFLAGS", value: "flag1")
        
        // Then
        XCTAssertEqual(subject.buildSettings["OTHER_LDFLAGS"] as? String, "flag1")
    }
```

**Pull request merged**